xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/XWiki.java

```diff
@@ -6679,6 +6679,9 @@ public String getURLContent(String surl, String username, String password, int t
         throws IOException
     {
         HttpClient client = getHttpClient(timeout, userAgent);
+        
+        // otherwise there is no authentification credentials will be sent
+        client.getParams().setAuthenticationPreemptive(true);
```

Comment:
This change might cause some problem to some use case. I assume forcing `Basic` (which is what you are doing here) could cause a problem when, for example, the endpoint actually requires something else like `Digest` (which seems supported by HttpClient 3 too and should work with the current code).

What might be safer to to introduce new #getURLContent API with a parameter to force Basic auth.